### PR TITLE
calendar

### PR DESCRIPTION
**Pull request by at-robins**
Some code refactoring and bug fixing is still required, but all the desired functionality is implemented working.

### CODE REVIEW

**Comment by at-robins** [comment]
At the time I implemented this I found it to be quite useful for everybody to have access to the schedule information, but I am not so sure anymore. Should I revert this, so the access is restricted to signed in users again?

**Comment by cgars** [comment]
I would keep it public!
Could you try it against [this](https://gist.github.com/cgars/e49510b875afe8981dd3710c6198c7c9)  (a bit more complex) schedule?

**Comment by at-robins** [comment]
Indeed there is a problem with tracks or sessions spanning multiple days. I need to manually break them down upon parsing.

**Comment by at-robins** [comment]
All the necessary fixes have been applied so the more complex JSON now works as intended.
Regarding the timezone support, currently independent of your timezone you will always see the same dates. Is this the intended functionality or should the dates be made timezone dependent?

**Comment by cgars** [comment]
So far it looks good!  Very good Job!!

Some short wishes:
 * Could the track/event/session titles be displayed next to the button and the timeslots?
* Could we have some sort of overview in the track view? eg. listing all events with there corresponding title and timeslots in there. Such that people could get a better idea whats happening.
* A bit more eye candy (different background color for tracks events etc.) would be nice. Its a bit hard to parse currently IMHO

With respect to timezones etc. Just keep it as is !

**Comment by at-robins** [comment]
All the requested changes are in. Is there something urgent that needs to be implemented/fixed for the first release version?

**Comment by cgars** [comment]
* Empty (eg. empty list) schedule should not be shown

**Comment by cgars** [comment]
* generally show only event fields that are set (eg.  Location, Abstract)

**Comment by cgars** [comment]
Integrated in master
